src/protocol/reveal_additive_binary.rs

```diff
@@ -0,0 +1,118 @@
+use crate::{
+    error::BoxError,
+    helpers::{fabric::Network, Direction},
+    protocol::{context::ProtocolContext, RecordId},
+};
+
+use serde::{Deserialize, Serialize};
+
+/// A message sent by each helper when they've revealed their own shares
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct RevealValue {
+    share: bool,
+}
+
+/// This implements a reveal algorithm for an additive binary secret sharing.
+/// As this is an additive sharing, each helper has just one boolean share
+/// As such, reveal requires each helper to send their share both left and right
+/// Put another way, each `P_i` sends `\[a\]_i` to `P_i+1` and `P_i-1`
+/// and then reconstructs a from` \[a\]_i`, `\[a\]_i+i` and `\[a\]i−1`.
+/// Input: Each helpers know their own secret shares
+/// Output: At the end of the protocol, all 3 helpers know a revealed (or opened) secret
+#[derive(Debug)]
+pub struct RevealAdditiveBinary {}
+
+impl RevealAdditiveBinary {
+    #[allow(dead_code)]
+    pub async fn execute<N: Network>(
+        ctx: ProtocolContext<'_, N>,
+        record_id: RecordId,
+        input: bool,
+    ) -> Result<bool, BoxError> {
+        let mut channel = ctx.mesh();
+
+        // Send share to helper to the left
+        channel
+            .send(
+                channel.identity().peer(Direction::Left),
+                record_id,
+                RevealValue { share: input },
+            )
+            .await?;
+
+        // Send share to helper to the right
+        channel
+            .send(
+                channel.identity().peer(Direction::Right),
+                record_id,
+                RevealValue { share: input },
+            )
+            .await?;
+
+        // Sleep until `helper's right` sends their share
+        let share_from_right: RevealValue = channel
+            .receive(channel.identity().peer(Direction::Right), record_id)
+            .await?;
+
+        // Sleep until `helper's left` sends their share
+        let share_from_left: RevealValue = channel
+            .receive(channel.identity().peer(Direction::Left), record_id)
+            .await?;
```

Comment:
It also really feels like this should be able to be done in parallel. As I mentioned above, I attempted to do so by using two channels (channel_left and channel_right), but it didn't work.